UHResidentInformationAPI.Tests/V1/Gateways/UHGatewayTests.cs

```diff
@@ -580,9 +560,87 @@ public void GetAllResidentsWhenAContactlinkRowsHasNoKey1AndKey2DoesNotBreakTheQu
             domainEntity.UPRN = address.UPRN;
             domainEntity.TenancyReference = tenancy1.TagRef;
 
-            var listOfPersons = _classUnderTest.GetAllResidents(address: "1 Hillman st");
+            var listOfPersons = _classUnderTest.GetAllResidents(null, 10, address: "1 Hillman st");
             listOfPersons.Count.Should().Be(1);
             listOfPersons.Should().ContainEquivalentOf(domainEntity);
         }
+
+        [Test]
+        public void GetAllResidentsWontReturnMoreRecordsThanTheLimit()
+        {
+            var person1 = AddPersonRecordToDatabase(houseRef: "123", personNo: 1);
+            var person2 = AddPersonRecordToDatabase(houseRef: "123", personNo: 2);
+
+            AddAddressRecordToDatabase(person1.HouseRef, address1: "1 Hillman st");
+            AddContactLinkForPersonToDatabase(person1.HouseRef, person1.PersonNo);
+
+            var peopleReturned = _classUnderTest.GetAllResidents(null, 1);
+
+            peopleReturned.Count.Should().Be(1);
+            peopleReturned[0].FirstName.Should().Be(person1.FirstName);
+            peopleReturned[0].LastName.Should().Be(person1.LastName);
+        }
+
+        [Test]
+        public void GetAllResidentsWillOffsetRecordsByTheCursor()
+        {
+            var persons = new List<Person>
+            {
+                AddPersonRecordToDatabase(houseRef: "123", personNo: 1),
```

Comment:
Hey Louise, just a quick note, it will be better if in the future we randomize those (e.g. using faker)

Reply:
Hmm - these tests cover the ordering in the gateway query (by HouseRef and PersonNo) as part of implementing the cursor. So I used these specific values here to highlight the role HouseRef plays in ordering and in the cursor.

However, I agree that I could improve next time, perhaps by using randomized values stored in a named list that makes the role of HouseRef clear in a better way 👍